ecl/network/v2/load_balancers/requests.go

```diff
@@ -0,0 +1,182 @@
+type CreateOpts struct {
+
+	// AvailabilityZone is one of the Virtual Server (Nova)’s availability zone.
+	AvailabilityZone string `json:"availability_zone,omitempty"`
+
+	// Description is description
+	Description string `json:"description,omitempty"`
+
+	// LoadBalancerPlanID is the UUID of Load Balancer Plan.
+	LoadBalancerPlanID string `json:"load_balancer_plan_id" required:"true"`
+
+	// Name is a human-readable name of the Load Balancer.
+	Name string `json:"name,omitempty"`
+
+	// The UUID of the project who owns the Load Balancer. Only administrative users
+	// can specify a project UUID other than their own.
+	TenantID string `json:"tenant_id,omitempty"`
+}
+
+// ToLoadBalancerCreateMap builds a request body from CreateOpts.
+func (opts CreateOpts) ToLoadBalancerCreateMap() (map[string]interface{}, error) {
+	b, err := eclcloud.BuildRequestBody(opts, "load_balancer")
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// Create accepts a CreateOpts struct and creates a new Load Balancer using the values
+// provided. You must remember to provide a valid LoadBalancerPlanID.
+func Create(c *eclcloud.ServiceClient, opts CreateOpts) (r CreateResult) {
+	b, err := opts.ToLoadBalancerCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = c.Post(createURL(c), b, &r.Body, &eclcloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// UpdateOpts represents the attributes used when updating an existing Load Balancer.
+type UpdateOpts struct {
+
+	// Description is description
+	DefaultGateway *string `json:"default_gateway,omitempty"`
+
+	// Description is description
+	Description *string `json:"description,omitempty"`
+
+	// LoadBalancerPlanID is the UUID of Load Balancer Plan.
+	LoadBalancerPlanID string `json:"load_balancer_plan_id,omitempty"`
+
+	// Name is a human-readable name of the Load Balancer.
+	Name *string `json:"name,omitempty"`
+}
+
+// ToLoadBalancerUpdateMap builds a request body from UpdateOpts.
+func (opts UpdateOpts) ToLoadBalancerUpdateMap() (map[string]interface{}, error) {
+	b, err := eclcloud.BuildRequestBody(opts, "load_balancer")
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// Update accepts a UpdateOpts struct and updates an existing Load Balancer using the
+// values provided.
+func Update(c *eclcloud.ServiceClient, id string, opts UpdateOpts) (r UpdateResult) {
+	b, err := opts.ToLoadBalancerUpdateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = c.Put(updateURL(c, id), b, &r.Body, &eclcloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// Delete accepts a unique ID and deletes the Load Balancer associated with it.
+func Delete(c *eclcloud.ServiceClient, id string) (r DeleteResult) {
+	_, r.Err = c.Delete(deleteURL(c, id), nil)
+	return
+}
+
+// IDFromName is a convenience function that returns a Load Balancer's ID,
+// given its name.
+func IDFromName(client *eclcloud.ServiceClient, name string) (string, error) {
```

Comment:
Can you write an unit test for this function or remove it if it is not used in terraform?

Reply:
OK, I wrote unit tests for the function IDFromName.